ArduPlane/mode_rtl.cpp

```diff
@@ -19,7 +19,21 @@ void ModeRTL::update()
     plane.calc_nav_pitch();
     plane.calc_throttle();
 
-    if (plane.g2.rtl_climb_min > 0) {
+    if (plane.g2.flight_options & FlightOptions::CLIMB_BEFORE_TURN) {
```

Comment:
this is a whole lot of redundant logic which makes it harder to read and maintain. All that is changing is the altitude threshold. Perhaps this could get reduced to:
```
    bool alt_threshold_reached = false;
    if (plane.g2.flight_options & FlightOptions::CLIMB_BEFORE_TURN) {
        // Climb to ALT_HOLD_RTL before turning. This overrides RTL_CLIMB_MIN.
        alt_threshold_reached = plane.current_loc.alt > plane.next_WP_loc.alt
    } else if (plane.g2.rtl_climb_min > 0) {
        /*
          when RTL first starts limit bank angle to LEVEL_ROLL_LIMIT
          until we have climbed by RTL_CLIMB_MIN meters
         */
        alt_threshold_reached = (plane.current_loc.alt - plane.prev_WP_loc.alt)*0.01 > plane.g2.rtl_climb_min
    } else {
        return;
    }

    if (!plane.rtl.done_climb && alt_threshold_reached) {
            plane.prev_WP_loc = plane.current_loc;
            plane.setup_glide_slope();
            plane.rtl.done_climb = true;
        }
    if (!plane.rtl.done_climb) {
            // Constrain the roll limit as a failsafe, that way if something goes wrong the plane will
            // eventually turn back and go to RTL instead of going perfectly straight. This also leaves
            // some leeway for fighting wind.
            plane.roll_limit_cd = MIN(plane.roll_limit_cd, plane.g.level_roll_limit*100);
            plane.nav_roll_cd = constrain_int32(plane.nav_roll_cd, -plane.roll_limit_cd, plane.roll_limit_cd);
    }




```

Reply:
Ah, you are correct, of course. I will change it shortly.